src/Sylius/Bundle/UiBundle/Twig/MergeRecursivelyExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UiBundle\Twig;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class MergeRecursivelyExtension extends AbstractExtension
+{
+    /**
+     * @return TwigFunction[]
+     */
+    public function  getFunctions(): array
+    {
+        return [
+            new TwigFunction(
+                'sylius_merge_recursively',
```

Comment:
I'd go with `sylius_merge_recursive` so that it mirrors `array_merge` and `array_merge_recursive`.